### PR TITLE
Add interface for resolving PayIDs

### PR DESCRIPTION
**Pull request by nkramer44**
## High Level Overview of Change
Introduces the `PayIDResolver` interface, whose implementations should resolve a `PayID` to an `HttpUrl`

### Context of Change
A `PayID` has to be resolved to a URL in order to make a request on the PayID server the `PayID` is hosted on.  

@sappenin has introduced [this RFC](https://github.com/xpring-eng/rfcs/blob/master/payid/src/spec/payid-discovery.md) for discovering/resolving PayIDs to URLs.  This PR is the first step in implementing this RFC.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After
New interface for resolving PayIDs.

## Test Plan
N/A

Two implementations of this interface.  One for WebFinger URL discovery ([automatic discovery](https://github.com/xpring-eng/rfcs/blob/master/payid/src/spec/payid-discovery.md#automated-mode)) and one for [manual resolution](https://github.com/xpring-eng/rfcs/blob/master/payid/src/spec/payid-discovery.md#manual-mode)

### CODE REVIEW

File: src/main/java/io/xpring/payid/PayIDResolver.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * Defines how to resolve a PayID to a URL.
+ *
+ * @see "https://github.com/xpring-eng/rfcs/blob/master/payid/src/spec/payid-discovery.md"
+ */
+@SuppressWarnings("checkstyle:AbbreviationAsWordInName")
+public interface PayIDResolver {
+
+  /**
+   * Resolves a {@link PayID} to it's endpoint URL.
```
Comment (amiecorso) [inline src/main/java/io/xpring/payid/PayIDResolver.java:14]: its not it's

File: src/main/java/io/xpring/payid/PayIDResolver.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+   *
+   * @param payID A {@link PayID} to resolve to an {@link HttpUrl}.
+   *
+   * @return The {@link HttpUrl} that {@code payID} resolves to.
+   *
+   * @see "https://github.com/xpring-eng/rfcs/blob/master/payid/src/spec/payid-discovery.md"
+   */
+  HttpUrl resolveHttpUrl(PayID payID);
```
Comment (sappenin) [inline src/main/java/io/xpring/payid/PayIDResolver.java:22]:
```suggestion
  Optional<HttpUrl> resolvePayIdUrl(PayID payID) throws PayIdDiscoveryException;
```

File: src/main/java/io/xpring/payid/PayIDDiscoveryExceptionType.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package io.xpring.payid;
+
+/**
+ * Types of {@link PayIDDiscoveryException}.
+ */
+@SuppressWarnings("checkstyle:AbbreviationAsWordInName")
+public enum PayIDDiscoveryExceptionType {
+  INVALID_PAY_ID,
```
Comment (keefertaylor) [inline src/main/java/io/xpring/payid/PayIDDiscoveryExceptionType.java:8]: nit: Alphabetize your enum cases? Unless you explicitly have some other ordering here.

**Comment by nkramer44** [comment]
Closing for now, since this seems to be out of scope for May 7th.

**Pull request closed without merging**